### ISSUE TITLE
ask about roof:levels where unset, start filling roof:levels=0 in countries where roof shape is not usually flat

### ISSUE THREAD

**Issue by matkoniecz** [streetcomplete/StreetComplete#4821]
Make `roof:levels` into a mandatory field in the building levels quest and re-ask building level quest for buildings that do have a `roof:shape` that is not `flat` but no`roof:levels` set. But:
- on new entry, prefill the roof levels field with "0". When tabbing into it / tapping on it, the whole text is selected so it can easily be replaced
- make the `roof:levels` field only mandatory (and prefill with 0 etc) in country where the roof shape is not usually flat (https://github.com/streetcomplete/countrymetadata/blob/master/data/roofsAreUsuallyFlat.yml)

this is copy of https://github.com/streetcomplete/StreetComplete/issues/4661#issuecomment-1431185989 and summary of https://github.com/streetcomplete/StreetComplete/issues/4661

**Comment by Helium314** [comment]
So the re-asking for non-flat roof levels would only happen in roofsAreUsuallyFlat countries?

Anyway, pre-filling (half) the answer might be dangerous, as SC then puts data that the user could have overlooked, e.g. if the user is used to not fill roof levels and doesn't notice the pre-filled 0. Though I guess mostly it will be fine.

**Comment by matkoniecz** [comment]
> So the re-asking for non-flat roof levels would only happen in roofsAreUsuallyFlat countries?

the re-asking for non-flat roof levels for buildings without `roof:levels` would happen in all countries

**Comment by matkoniecz** [comment]
> Anyway, pre-filling (half) the answer might be dangerous, as SC then puts data that the user could have overlooked, e.g. if the user is used to not fill roof levels and doesn't notice the pre-filled 0.

SC is doing the same, just not tagging 0 explicitly

**Comment by Discostu36** [comment]
> the re-asking for non-flat roof levels for buildings without `roof:levels` would happen in all countries

I am not convinced that in these cases the field should be prefilled with 0 because of:
- High likelihood of users adding wrong data accidentally
- Low likelihood of non-flat buildings having zero roof levels

**Comment by westnordost** [comment]
Ok, no pre-filling with 0 then.

**Comment by westnordost** [comment]
Hm, I'd say that if you can't see the roof, you neither can say with certainty whether the building levels are correct in some cases: There could be no roof, but there could be a penthouse (of several levels) on top which you can't see from below because it is offset towards the center of the building.

**Comment by grrrrr** [comment]
@westnordost 

You can see the roof but it's not necessarily obvious that there is a level in it.   For example my house have 2 building levels and 1 in the roof, the roof is hipped but you can not tell this from the street, only when looking at the back of the house which most people can not do.


Also at the moment,  even with new unmapped buildings you must supply a roof level,  is this correct?   With as I described above, it's greatly hampering mapping of building levels accurately.

**Comment by Discostu36** [comment]
In my experience there are many cases where you can tell building levels with high certainty, but not roof levels.

Just one of hundreds of examples from my hometown:

![Image]](https://upload.wikimedia.org/wikipedia/commons/thumb/5/52/Wei%C3%9Fenburgstr._12%2C_Kiel.jpg/463px-Wei%C3%9Fenburgstr._12%2C_Kiel.jpg)

This is why I [suggested](https://github.com/streetcomplete/StreetComplete/issues/4661#issuecomment-1351556400) that it might be better to have a separate, disabled roof level quest.

**Comment by westnordost** [comment]
Safe to say that if there'd be an overlay for building levels, it would be no problem to allow incomplete input.

Also, note that the roof levels were optional were never intended to allow to not specify also the roof levels, but allow to not specify them **if there are no roof levels**, due to the (then, maybe still) prominent opinion that `roof:levels=0` is the default and does not need to be tagged if there are no roof levels.

I acknowledge your argument, oftentimes you can see the building levels, but not the roof levels. But oh well, to be honest, this issue has been such a long discussion and so much forth and back that I do not recall the exact reasons for the current solution.

If I remember correctly, another reason for the current solution was that the building levels quest is disabled by default. So the assumption is that people who have it activated are rather interested in supplying the information in a complete, comprehensive form. After all, arguably, for what the data is used, 3D rendering and statistical analysis ([such as for natural disaster resilience](https://media.ccc.de/v/fossgis2023-23821-erdbeben-und-openstreetmap)), it is somewhat important to have this data in a complete form. After all, if the roof level data is missing, any data consumer will actually have to assume `roof:levels=0`. But if this assumption is widely wrong (because people only added the non-roof levels in StreetComplete), the derived data becomes somewhat wrong too.
In a nutshell, I would advise against mapping only `building:levels` and not `roof:levels`.

**Comment by grrrrr** [comment]
> I acknowledge your argument

hi @westnordost please don't think of this as an argument :(  just a discussion / suggestion as we all want the best for both StreetComplete and OSM as a whole :)

> After all, arguably, for what the data is used, 3D rendering and statistical analysis ([such as for natural disaster resilience](https://media.ccc.de/v/fossgis2023-23821-erdbeben-und-openstreetmap), minute 21 and after), it is somewhat important to have this data in a complete form. After all, if the roof level data is missing, any data consumer will actually have to assume roof:levels=0

I think this is the wrong way to look at it,   yes statistics and 3D rendering are important but couple of thing

1.  in my above example (2 building level, 1 roof level, gabled roof)  This still3D  renders correctly if the roof:level is left out.  
2. for natural disaster resilience  if roof level has not been mapped but building level has, you know there are at least two levels in the building and as there is no `roof:levels` tag it's an unknown and contingency can be made.     But in the new way street complete does it,  there is now the possibility of bad data being put into OSM,  if people have to put in roof:level=0  if they do not know.  Now someone goes to the building expecting no roof level but discovers there are one.

Both data are useful to have but should not be reliant on both being put in.


My suggestion would be to revert to the way it was,  OSM becomes more complete even if it's just building levels you can add building level without roof level - OSM becomes more complete (and accurate) with just building levels.   

Longer term, @Discostu36 [comment](https://github.com/streetcomplete/StreetComplete/issues/4661#issuecomment-1351556400) sounds like a good solution.  And the roof:levels task could be disabled by default as it's much harder to accurately do, unlike the majority of building levels

**Comment by westnordost** [comment]
>> I acknowledge your argument

> hi @westnordost please don't think of this as an argument :(

Sorry, that was Germanenglish. Better translation would be "I understand your point".

>  if people have to put in roof:level=0 if they do not know

Well, that would be vandalism. What user would think that "0" means "I don't know"?

**Comment by grrrrr** [comment]
> Well, that would be vandalism. What user would think that "0" means "I don't know"?

I can see it possibly happening,  after all  Street Complete is designed for all levels of OSM editor old and new and not necessarily knowing the error of mistagging something. it's they are just making an assumption based on what they can see on the ground.   Not necessarily vandalism but they are happy to be able to contribute to the database

Decoupling the requirement to fill-in roof:levels to be able to submit building:levels reduces this liklihood though.

It's a bit like some of the other tags, you are merely adding information you can glean where possible e.g. you don't force users to define sidewalks or bike paths at the same time as doing car lanes even though they would be tagged on the same way.  I can't think of any other two sets of separate (but complimentary) tags that StreetComplete requires the user to fill in at the same time to be able to submit one of them (correct me if I am wrong).
